Lib/test/test_enum.py

```diff
@@ -1776,19 +1776,12 @@ def _missing_(cls, item):
                     # trigger not found
                     return None
         self.assertIs(Color('three'), Color.blue)
-        self.assertRaises(ValueError, Color, 7)
-        try:
+        with self.assertRaises(ValueError):
+            Color(7)
+        with self.assertRaises(TypeError):
             Color('bad return')
-        except TypeError as exc:
-            self.assertTrue(isinstance(exc.__context__, ValueError))
-        else:
-            raise Exception('Exception not raised.')
-        try:
+        with self.assertRaises(ZeroDivisionError):
             Color('error out')
-        except ZeroDivisionError as exc:
-            self.assertTrue(isinstance(exc.__context__, ValueError))
```

Comment:
Hm, I hadn't noticed but this test would (still) be affected by the fix.  And it seems reasonable to keep : \

Instead of changing when context is added to the exception, could the default implementation of `_missing` be changed to return `None`?  Or would that break something?

Reply:
I think that would break logic because `None` might be as value of enum:
```Python
from enum import Enum

class Color(Enum):
    BLUE = 'blue'
    RED = 'red'
    UNDEFINED = None
```

Reply:
But in that example `_missing` would need to return `cls.UNDEFINED` instead of `None`, because it has to return an instance of `cls`.

Reply:
Maybe you're right) I will try it

Reply:
The original verification here should be kept. It's checking that if using `_missing_` explodes when searching for a value, the final result is the ValueError (of the bad value) and the ZeroDivisionError (of the internal problem).